cmd/minishift/cmd/status.go

```diff
@@ -19,29 +19,84 @@ package cmd
 import (
 	"fmt"
 	"os"
+	"strings"
+	"text/template"
 
 	"github.com/docker/machine/libmachine"
-	"github.com/spf13/cobra"
-
+	"github.com/docker/machine/libmachine/state"
 	"github.com/minishift/minishift/pkg/minikube/cluster"
 	"github.com/minishift/minishift/pkg/minikube/constants"
+	openshiftVersion "github.com/minishift/minishift/pkg/minishift/openshift/version"
+	profileActions "github.com/minishift/minishift/pkg/minishift/profile"
 	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/spf13/cobra"
 )
 
+var statusFormat = `Minishift:  {{.MinishiftStatus}}
+OpenShift:  {{.ClusterStatus}}
+Profile:    {{.ProfileName}}
+DiskUsage:  {{.DiskUsage}}
+`
+
+type Status struct {
+	MinishiftStatus string
+	ClusterStatus   string
+	ProfileName     string
+	DiskUsage       string
+}
+
 // statusCmd represents the status command
 var statusCmd = &cobra.Command{
 	Use:   "status",
 	Short: "Gets the status of the local OpenShift cluster.",
 	Long:  `Gets the status of the local OpenShift cluster.`,
-	Run: func(cmd *cobra.Command, args []string) {
-		api := libmachine.NewClient(constants.Minipath, constants.MakeMiniPath("certs"))
-		defer api.Close()
+	Run:   runStatus,
+}
+
+func runStatus(cmd *cobra.Command, args []string) {
+	api := libmachine.NewClient(constants.Minipath, constants.MakeMiniPath("certs"))
+	defer api.Close()
+
+	host, err := api.Load(constants.MachineName)
+	if err != nil {
 		s, err := cluster.GetHostStatus(api)
 		if err != nil {
 			atexit.ExitWithMessage(1, fmt.Sprintf("Error getting cluster status: %s", err.Error()))
 		}
-		fmt.Fprintln(os.Stdout, s)
-	},
+		atexit.ExitWithMessage(0, s)
+	}
+
+	openshiftStatus := state.None.String()
+	profileName := state.None.String()
+	diskUsage := state.None.String()
+
+	vmStatus, err := cluster.GetHostStatus(api)
+	if err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error getting cluster status: %s", err.Error()))
+	}
+
+	if vmStatus == state.Running.String() {
+		openshiftVersion, err := openshiftVersion.GetOpenshiftVersion(host)
+		openshiftStatus = "Stopped"
+		if err == nil {
+			openshiftStatus = fmt.Sprintf("Running (%s)", strings.Split(openshiftVersion, "\n")[0])
+		}
```

Comment:
@praveenkumar Seems like we have taken a short cut to find out if the OpenShift is running or not. I would suggest we implement a function IsOpenshiftRunning.

Reply:
Agree :+1:

Reply:
@LalatenduMohanty It is not a sortcut if you check the implementation of this method it actually run the command on the origin container name. If think we should have a different method which tells openshift status then what docker command you have in mind? I still thinks this is clean.

Reply:
@LalatenduMohanty 
```
docker@minishift:~$ docker exec origin openshift version
openshift v3.6.0+c4dd4cf
kubernetes v1.6.1+5115d708d7
etcd 3.2.1
docker@minishift:~$ docker stop origin
origin
docker@minishift:~$ docker exec origin openshift version
Error response from daemon: Container e31d14e98b38202279fdbd04f1211859674c3a7210bada2d6924ae216518746c is not running
```

Reply:
@praveenkumar @gbraad Either we should use `oc cluster status` to see if Openshift instance is running or not. Here is the cluster up implementation https://github.com/openshift/origin/blob/master/pkg/oc/bootstrap/docker/status.go

Reply:
We can do it using `ocRunner` .  IMO we should be able to use the function to get the status without worrying about the VM state as the function will handle the same.

Reply:
@LalatenduMohanty There is a reason why we are using version because if I use `ocRunner` there is no way I can get running version of the cluster using `oc` commands. if you look the output of status command when running is look below.

```
MiniShift:  Running
OpenShift:  Running (openshift v3.6.0+c4dd4cf)  => here running openshift version coming from version fuctions.
Profile:    minishift
Storage:    used 7% of 27.3G
```

Reply:
As discussed with @praveenkumar , I am ok with current implemented.